### PR TITLE
Sql

### PR DESCRIPTION
**Pull request by IsraaSulaiman**
### New Features made in this pull request:
1. dbBuild.sql: #62 
 It contains 5 tables: users, plans, categories, category_plan, expenses. The name of columns are based on our schema #7 
2. dbConnection.js: 
It contains a function to build connection with the database server. #62 
3. dbBuild.js:  
 It contains a function to rebuild the SQL according to our schema. This function will be used basically in testing and before deployment. #62 

### Edits:

1. The script in package.json has been edited. Different configurations have been added for different environments: development, testing, and production environments. 

### Bug Fix:

1. This pull request also includes some bug fixes in the SQL and other issues with eslint.

**There was a conflict in package.json. It was resolved.**

### CODE REVIEW

File: src/database/dbConnection.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+require('dotenv').config();
```
Comment (engshorouq) [inline src/database/dbConnection.js]: this line is already called in app.js 
remove it

Comment (IsraaSulaiman) [reply src/database/dbConnection.js]: I removed it

**Review by mohammedmh — APPROVED** [review]

**Review by KhaderMurtaja — APPROVED** [review]

File: src/database/dbBuild.sql [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  (
+    id SERIAL PRIMARY KEY,
+    plan_id INTEGER REFERENCES plans(id),
+    cat_id INTEGER REFERENCES categories(id),
+    price FLOAT NOT NULL,
+    date TEXT NOT NULL,
+    description TEXT
+  );
+
+  INSERT INTO categories
```
Comment (asmaa-ezz) [inline src/database/dbBuild.sql]:
```suggestion
  INSERT INTO categories
```
in file **db Build.sql** mustn't to add any query
just to write schema

File: src/database/dbBuild.sql [hunk 3]
```diff
@@ -0,0 +1,60 @@
+BEGIN;
+
+  DROP TABLE IF EXISTS users, categories, plans, category_plan, expenses;
+
+  CREATE TABLE users
```
Comment (asmaa-ezz) [inline src/database/dbBuild.sql]:
```suggestion
  CREATE TABLE users
```
Table names should preferably be a single name
but the name `user` the reserved word in sql
so you can to solve this write the table name in quotations

```SQL
 CREATE TABLE "user"
```

**Review by engshorouq — APPROVED** [review]

File: src/database/dbConnection.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const { Pool } = require('pg');
+const url = require('url');
+
+let dbUrl = process.env.DB_LOCAL_URL;
+if (process.env.NODE_ENV === 'pro') {
+  dbUrl = process.env.DB_HEROKU_URL;
+}
```
Comment (asmaa-ezz) [inline src/database/dbConnection.js]: {Error in ESLint }
Closing curly brace does not appear on the same line as the subsequent block.

File: src/database/dbConnection.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+const { Pool } = require('pg');
+const url = require('url');
+
+let dbUrl = process.env.DB_LOCAL_URL;
```
Comment (asmaa-ezz) [inline src/database/dbConnection.js]:
```suggestion
let dbUrl = process.env.DB_LOCAL_URL;
```
if the process.env.DB_LOCAL_URL equal ''undefined" what the happen?
and if I do not have database ?

File: src/database/dbConnection.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+if (process.env.NODE_ENV === 'pro') {
+  dbUrl = process.env.DB_HEROKU_URL;
+}
+else if (process.env.NODE_ENV === 'test') {
+  dbUrl = process.env.DB_TESTING_URL;
+}
+else {
+  dbUrl = process.env.DB_LOCAL_URL;
```
Comment (asmaa-ezz) [inline src/database/dbConnection.js:10]:
```suggestion
  dbUrl = process.env.DB_LOCAL_URL;
```
What do I need this sentence has given the variable the same value as the previous one

**Review by asmaa-ezz — COMMENTED** [review]
Solve the problem and conflict

**Pull request merged**